src/main/tut/docs/module9.md

```diff
@@ -2,7 +2,623 @@
 layout: docs
 title: "Functional programming"
 section: "docs"
-author: author
+author: M.C. Oscar Vargas Torres
 ---
 
 # Module 9: Functional programming
+
+## Introduction and prerequisites
+
+### For comprehensions
+
+The following resources could be helpful:
+
+- Module 7 material
+- [Essential Scala], chapter 5
+- [Better monadic for plugin]
+- [For comprehensions] from Sam Halliday (@fommil)
+
+[Essential Scala]: https://underscore.io/books/essential-scala/
+[Better monadic for plugin]: https://github.com/oleg-py/better-monadic-for
+[For comprehensions]: https://leanpub.com/fpmortals/read#leanpub-auto-for-comprehensions
+
+### Using more precise types
+
+If you have been practicing with Scala, it is probable that you have noticed that using precise
+types reduces bugs. For example, given the following `Person` definition, you could possibly define
+instances that do not model reality:
+
+```scala
+case class Person(name: String, age: Int)
+// Bad input
+val unrealPerson = Person("   ", -30)
+```
+
+because, arguably, a person's name cannot be "   ", and all ages are non-negative, and are
+reasonably (arbitrarily) bounded by 127 (or 200, if you wish).
+
+If it is possible, strive to use types that completely eliminate trivial bugs/errors. A simple way
+to define more precise/refined types, is by using [Refined]. The above example could be rewritten
+with:
+
+```scala
+package io.github.oscarvarto
```

Comment:
What if you define an "organizational" package like: `com.intersys.edge.scalananodegree`. Through commits inspection, people still will know that you're the author of this content.

Reply:
Yeah, it was a copy/paste mistake.

Reply:
Fixed.